tuned/plugins/plugin_scheduler.py

```diff
@@ -136,8 +136,292 @@ def get_priority_max(self, sched):
+	====
+	----
+	[scheduler]
+	cgroup_mount_point=/sys/fs/cgroup/cpuset
+	cgroup_mount_point_init=1
+	cgroup_groups_init=1
+	cgroup_for_isolated_cores=group
+	cgroup.group1=2
+	cgroup.group2=0,2
+	
+	group.ksoftirqd=0:f:2:cgroup.group1:ksoftirqd.*
+	ps_blacklist=ksoftirqd.*;rcuc.*;rcub.*;ktimersoftd.*
+	isolated_cores=1
+	----
+	Cgroup `group1` has the affinity set to CPU 2 and the cgroup `group2`
+	to CPUs 0,2. Given a 4 CPU setup, the [option]`isolated_cores=1`
+	option causes all processes/threads to be moved to CPU
+	cores 0,2-3. Processes/threads that are blacklisted by the
+	[option]`ps_blacklist` regular expression will not be moved.
+	
+	The scheduler plug-in will isolate the specified core by writing the
+	CPU affinity 0,2-3 to the `cpuset.cpus` control file of the `group`
+	and move all the matching processes/threads to this cgroup.
+	====
+	Option [option]`cgroup_ps_blacklist` allows excluding processes
+	which belong to the blacklisted cgroups. The regular expression specified
+	by this option is matched against cgroup hierarchies from
+	`/proc/PID/cgroups`. Cgroups v1 hierarchies from `/proc/PID/cgroups`
+	are separated by commas ',' prior to regular expression matching. The
+	following is an example of content against which the regular expression
+	is matched against: `10:hugetlb:/,9:perf_event:/,8:blkio:/`
+	+
+	Multiple regular expressions can be separated by semicolon ';'. The
+	semicolon represents a logical 'or' operator.
+	+
+	.Cgroup-based exclusion of processes from the scheduler
+	====
+	----
+	[scheduler]
+	isolated_cores=1
+	cgroup_ps_blacklist=:/daemons\b
+	----
+	
+	The scheduler plug-in will move all processes away from core 1 except processes which
+	belong to cgroup '/daemons'. The '\b' is a regular expression
+	metacharacter that matches a word boundary.
+	
+	----
+	[scheduler]
+	isolated_cores=1
+	cgroup_ps_blacklist=\b8:blkio:
+	----
+	
+	The scheduler plug-in will exclude all processes which belong to a cgroup
+	with hierarchy-ID 8 and controller-list blkio.
+	====
+	Recent kernels moved some `sched_` and `numa_balancing_` kernel run-time
+	parameters from `/proc/sys/kernel`, managed by the `sysctl` utility, to
+	`debugfs`, typically mounted under `/sys/kernel/debug`.  TuneD provides an
+	abstraction mechanism for the following parameters via the scheduler plug-in:
+	[option]`sched_min_granularity_ns`, [option]`sched_latency_ns`,
+	[option]`sched_wakeup_granularity_ns`, [option]`sched_tunable_scaling`,
+	[option]`sched_migration_cost_ns`, [option]`sched_nr_migrate`,
+	[option]`numa_balancing_scan_delay_ms`,
+	[option]`numa_balancing_scan_period_min_ms`,
+	[option]`numa_balancing_scan_period_max_ms` and
+	[option]`numa_balancing_scan_size_mb`.
+	Based on the kernel used, TuneD will write the specified value to the correct
+	location.
+	+
+	.Set tasks' "cache hot" value for migration decisions.
+	====
+	----
+	[scheduler]
+	sched_migration_cost_ns=500000
+	----
+	On the old kernels, this is equivalent to:
```

Comment:
> Do we need to define "old" kernels?

I do not believe so.  It would be easy if every distro used a pristine kernel X.Y.Z without any mods.  But they do not.  I know that "old" is vague, but I'm not sure we can do any better.